File: Exercism Codes/Java/CollatzCalculator.java
class CollatzCalculator {

    int computeStepCount(int start) {
        if(start <= 0) throw new IllegalArgumentException("Only natural numbers are allowed");
        int count = 0;
        while(start != 1) {
            start = start % 2 == 0 ? start / 2 : (start * 3) + 1;
            count++;
        }
        return count;
    }

}

File: Exercism Codes/Kotlin/CollatzCalculator.kt
object CollatzCalculator {
    fun computeStepCount(start: Int): Int {
        require(start > 0)
        var s = start
        var step = 0
        while (s > 1) {
            s = if (s % 2 == 0) s / 2 else (s * 3) + 1
            step++
        }
        return step
    }
}
